lms/djangoapps/shoppingcart/models.py

```diff
@@ -1337,6 +1372,33 @@ def add_coupon_redemption(cls, coupon, order, cart_items):
 
         return is_redemption_applied
 
+    @classmethod
+    def get_top_discount_codes_used(cls, course_id):
+        """
+        Returns the top five discount code used.
+
+        QuerySet = [
+            {
+                'coupon__percentage_discount': 22,
+                'coupon__code': '12',
+                'coupon__used_count': '2',
+            },
+            {
+                ...
+            }
+        ]
+        """
+        return cls.objects.filter(order__status='purchased', coupon__course_id=course_id).values(
+            'coupon__code', 'coupon__percentage_discount'
+        ).annotate(coupon__used_count=Count('coupon__code'))
```

Comment:
One thing, this query will not show discount codes that haven't been used at all. I'm noticing it on the sandbox (I just created a coupon but not used). It's probably OK for now to not display 0 used coupons, but might be something we look into later on.